src/internet_identity/src/http.rs

```diff
@@ -253,6 +214,61 @@ fn security_headers() -> Vec<HeaderField> {
     ]
 }
 
+/// Full content security policy delivered via HTTP response header.
+///
+/// This policy also includes the `frame-ancestors` directive in addition to the policies included in the HTML `meta` tag.
+/// We deliver the CSP by header _and_ meta tag because the headers are not yet certified.
+fn content_security_policy_header() -> String {
+    let meta_policy = content_security_policy_meta();
+    format!("{meta_policy} frame-ancestors 'none';").to_string()
+}
+
+/// Stripped down content security policy for the HTML `meta` tag, where not all directives are supported.
+///
+/// The sha256 hash matches the inline script in index.html. This inline script is a workaround
+/// for Firefox not supporting SRI (recommended here https://csp.withgoogle.com/docs/faq.html#static-content).
+/// This also prevents use of trusted-types. See https://bugzilla.mozilla.org/show_bug.cgi?id=1409200.
+///
+/// script-src 'unsafe-eval' is required because agent-js uses a WebAssembly module for the
+/// validation of bls signatures.
+/// There is currently no other way to allow execution of WebAssembly modules with CSP.
+/// See https://github.com/WebAssembly/content-security-policy/blob/main/proposals/CSP.md.
+///
+/// script-src 'unsafe-inline' https: are only there for backwards compatibility and ignored
+/// by modern browsers.
+///
+/// connect-src https://*.ic0.app is required in order for II to be able to fetch the
+/// /.well-known/ii-alternative-origins path of authenticating canisters setting a derivationOrigin.
+///
+/// style-src 'unsafe-inline' is currently required due to the way styles are handled by the
```

Comment:
I _think_ we can remove this now that we load the css via `link` (for another PR)